### ISSUE TITLE
Implement a "safe mode" for blinding

### ISSUE THREAD

**Issue by ntessore** [euclidlib/euclidlib#39]
Euclidlib should implement a "safe mode" that prevents products with the LE3 blinding flag set to "don't read" from being used accidentally.

CC @sfarrens: Having this feature means our blinding flag needs to be 3 state: unblinded, blinded, and "safe", e.g., for mixing matrices. Is it?

**Comment by gcanasherrera** [comment]
Thanks for this enhancement. Just for me to understand, wouldn't it make more sense that the name of the file is representative of the data product it contains so that people are aware of what they are opening?

**Comment by sfarrens** [comment]
@ntessore I think what Sergio implemented was purely binary (unblinded and blinded). If a third option was needed, we would probably need to update the DM, but I think this could be done.

@gcanasherrera we can look into this. It is probably possible to added something like `_blinded` to the FITS filenames of the products that pass through the blinder.

**Comment by ntessore** [comment]
@gcanasherrera That's true, but we don't know yes what names come out of the SAS eventually, and accidents can still happen.

In any case, since there is the BLINDING flag in the FITS, and euclidlib is meant to be the access point to the FITS, it would make sense for euclidlib to provide access to the blinding flag.

If euclidlib provides access to the blinding flag, then I would naturally write code that is meant to operate on blinded products in such a way that it checks whether the blinding flag is set. It hence makes sense to me that this functionality is absorbed into euclidlib.

In practice, I can imagine an interface like the following. All of the blinding stuff should be opt-in, hence there is some sort of general "blinding-awareness" setup.

```py
el.enforce_blinding(True)

# attempt to read a non-blinded product
cls = el.photo.angular_power_spectra("oops.fits")
# BlindingError: oops.fits: file is not blinded

# but maybe you really wanted to read that file
with el.ignore_blinding:
    cls = el.photo.angular_power_spectra("not_oops.fits")
    # ok
```

@sfarrens Ok, let's look into it, because that's pretty much independent of this issue here IMO. We want to be able to easily disambiguate "not blinded" from "blinding does not apply".

**Comment by gcanasherrera** [comment]
Thanks to both for your suggestions. Indeed, the discussion about the name of the files should take place within the EC SGS infrastructure sphere. 

With respect to this blinding mode, I got convinced after seeing @ntessore suggestion, which I think can work well if we explain it nicely in the docs. 

I am wondering now if it would be worth adding some metadata to the read euclidlib python dictionaries where we save the name of the product and other flags, such as if the product is blinded or not. Shall I open a discussion topic to iterate over this?

**Comment by gcanasherrera** [comment]
Given that we are approaching to the TR1 cycle, do we want to push/further investigate this @sfarrens @ntessore ?

**Comment by gcanasherrera** [comment]
HI @sfarrens , thanks.

I think that having in the header something like "blinded" would be nice, so we can raise a warning with euclidlib about the product being blinded when opened up.
